File: clojure/raindrops/src/raindrops.clj
(ns raindrops)

(def factors {3 "Pling"
              5 "Plang"
              7 "Plong"})

(defn check-factor [div response num]
  (when (zero? (rem num div))
    response))

(defn convert [num]
  (->> (map #(check-factor (key %) (val %) num) factors)
       (apply str)
       (#(if (empty? %) (str num) %))))

File: fsharp/raindrops/Raindrops.fs
module Raindrops

let rainDropSounds =
    [ (3, "Pling")
      (5, "Plang")
      (7, "Plong") ]

let convert (number: int) : string =
    rainDropSounds
    |> List.filter (fun (f, _) -> number % f = 0)
    |> List.map snd
    |> List.fold (+) ""
    |> function
    | "" -> $"{number}"
    | s -> s
